src/platform/test.rs

```diff
@@ -412,6 +414,47 @@ fn receiver_set() {
 }
 
 #[test]
+#[cfg(not(any(feature = "force-inprocess", target_os = "windows", target_os = "android")))]
+fn receiver_set_eintr() {
+    let (start_server, start_name) = OsIpcOneShotServer::new().unwrap();
+    let (end_server, end_name) = OsIpcOneShotServer::new().unwrap();
+    let child_pid = unsafe {
+        fork(|| {
+            let (tx0, rx0) = platform::channel().unwrap();
+            let mut rx_set = OsIpcReceiverSet::new().unwrap();
+            let rx_id = rx_set.add(rx0).unwrap();
+            // Let the parent know we're ready
+            let tx1 = OsIpcSender::connect(start_name).unwrap();
+            tx1.send(b" Ready! ", vec![OsIpcChannel::Sender(tx0)], vec![]).unwrap();
+            // Send the result of the select back to the parent
+            let result = rx_set.select().unwrap();
+            let mut result_iter = result.into_iter();
+            let (id, data, _, _) = result_iter.next().unwrap().unwrap();
+            assert_eq!(rx_id, id);
+            assert_eq!(b"Test".as_ref(), &*data);
+            assert!(result_iter.next().is_none());
+            let tx2 = OsIpcSender::connect(end_name).unwrap();
+            tx2.send(b"Success!", vec![], vec![]).unwrap();
+        })
+    };
+    // Wait until the child is ready
+    let (_, res, mut channels, _) = start_server.accept().unwrap();
+    assert!(res == b" Ready! ");
+    assert!(channels.len() == 1);
```

Comment:
True, this is a complicated enough test, so IMO any removal of unnecessary asserts or code is a win.